File: app/src/main/java/com/ilifesmart/java/Button.java
package com.ilifesmart.java;

import android.util.Log;
import com.ilifesmart.kotlin.State;
import com.ilifesmart.kotlin.View2;
import org.jetbrains.annotations.NotNull;

public class Button implements View2 {
    public static final String TAG = "JavaButton";

    @NotNull
    @Override
    public State getCurrentState() {
        Log.d(TAG, "getCurrentState: ");
        return new ButtonState();
    }

    @Override
    public void restoreState(@NotNull State state) {

    }

    public class ButtonState implements State {

    }
}

File: app/src/main/java/com/ilifesmart/kotlin/Button.kt
package com.ilifesmart.kotlin

class Button2: View2 {
    override fun getCurrentState(): State {
        println("getCurrentState")
        return ButtonState()
    }

    override fun restoreState(state: State) {
        println("restoreState--")
    }

    // 内部类(保存引用)持有外部类引用及访问方法
    inner class ButtonState2: State {
        fun getButton2(): Button2 = this@Button2
    }

    // 内部嵌套类(不保存引用)
    class ButtonState: State {
    }
}